freqtrade/plugins/protections/iprotection.py

```diff
@@ -80,6 +81,45 @@ def lookback_period_str(self) -> str:
         else:
             return f"{self._lookback_period} {plural(self._lookback_period, 'minute', 'minutes')}"
 
+    @property
+    def unlock_at_str(self) -> Union[str, None]:
+        """
+        Output configured unlock time
+        """
+        if self.unlock_at:
+            return self.unlock_at.strftime("%H:%M")
+        return None
+
+    def set_unlock_at_as_stop_duration(self) -> None:
+        """
+        Calculates the stop_duration based on the unlock_at protection config value and sets it.
+        """
+        if "unlock_at" in self._protection_config:
+            self._stop_duration = self.calculate_unlock_at()
+            return None
+
+        logger.warning(
+            "Couldn't update the stop duration, because unlock_at is not set in the "
+            "protection config."
+        )
```

Comment:
I don't think you understood me correctly.

if you aim to use `unlock_at` - then there is no `_stop_duration`. 
It'll change per "protection trigger" - but then it's unnecessary / pointless (up to being missleading) to set this - as it'll change with every call.

With that in mind what point would this whole method serve? 

you can't run into the warning - if you do - you didn't validate the configuration properly.

Reply:
Yeah makes sense, thanks.

Reply:
I thought the stop_duration must be updated from the protections. That's why, I added a setter class which I could call from the protections.